torch/quantization/quantize.py

```diff
@@ -216,6 +217,14 @@ def prepare(model, inplace=False, allow_list=None,
         prehook: observer we want to add to forward_pre_hook
     """
     torch._C._log_api_usage_once("quantization_api.quantize.prepare")
+    if is_custom_observed_module_class(model):
+        if inplace:
+            warnings.warn("'inplace' does not apply to the custom modules")
+        model = get_observed_custom_module_class(model).from_float(model)
+        return prepare(
```

Comment:
what's the context for using recursion here, and for doing it with inplace=True?  Can we avoid either/both?

Reply:
The process here is that the `get_observed_custom_module_class` replaces the model with a different implementation. After that we can prepare the newly instantiated model. The motivation for `inplace` is that because the newly created model is not the same as the original, there is no need to deepcopy it.